torch/csrc/jit/tensorexpr/loopnest.h

```diff
@@ -412,6 +417,13 @@ class TORCH_API LoopNest {
   // }
   static void compressBuffer(Buf* buf, Stmt* stmt);
 
+  // Compresses all buffers in the given statement.
+  //
+  // NOTE: This API assumes that there are no accesses to buffers outside
+  // the given statement. So, this should be called with the entire
+  // kernel statement to avoid incorrect buffer compressions.
```

Comment:
It might be a good idea to add a check to IR verifier that would detect invalidly compressed buffers (not for this PR, but might be worth mentioning as a TODO).

Reply:
Added a TODO